src/Features/Lsif/GeneratorTest/FoldingRangeTests.vb

```diff
@@ -0,0 +1,115 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.VisualStudio.LanguageServer.Protocol
+Imports Roslyn.Test.Utilities
+Imports Roslyn.Utilities
+
+Namespace Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.UnitTests
+    <UseExportProvider>
+    Public NotInheritable Class FoldingRangeTests
+        Private Const TestProjectAssemblyName As String = "TestProject"
+
+        ' Expected 'FoldingRangeKind' argument would likely change for some of the tests when
+        ' https://github.com/dotnet/roslyn/projects/45#card-20049168 is implemented.
+        <Theory>
+        <InlineData("
+class C{|foldingRange:
+{
+}|}", Nothing)>
+        <InlineData("
+class C{|foldingRange:
+{
+    void M(){|foldingRange:
+    {
+        for (int i = 0; i < 10; i++){|foldingRange:
+        {
+            M();
+        }|}
+    }|}
+}|}", Nothing)>
+        <InlineData("
+{|foldingRange:#region
+#endregion|}", Nothing)>
+        <InlineData("
+using {|foldingRange:System;
+using System.Linq;|}", FoldingRangeKind.Imports)>
+        <InlineData("
+using {|foldingRange:S = System.String; 
+using System.Linq;|}", FoldingRangeKind.Imports)>
+        <InlineData("
+{|foldingRange:// Comment Line 1
+// Comment Line 2|}", Nothing)>
+        Public Async Function TestFoldingRanges(code As String, rangeKind As FoldingRangeKind?) As Task
+            Using workspace = TestWorkspace.CreateWorkspace(
+                    <Workspace>
+                        <Project Language="C#" AssemblyName=<%= TestProjectAssemblyName %> FilePath="Z:\TestProject.csproj" CommonReferences="true">
+                            <Document Name="A.cs" FilePath="Z:\A.cs">
+                                <%= code %>
+                            </Document>
+                        </Project>
+                    </Workspace>)
+
+                Dim annotatedLocations = AbstractLanguageServerProtocolTests.GetAnnotatedLocations(workspace, workspace.CurrentSolution)
```

Comment:
I appreciate this is annotating similar to how the LSP tests do, which is nice and consistent if somebody gets to debug both parts.